chain/near.ts

```diff
@@ -0,0 +1,416 @@
+    }
+
+    static fromTransfer(input: TransferAction): ActionValue {
+      return new ActionValue(ActionKind.TRANSFER, changetype<u32>(input))
+    }
+
+    static fromStake(input: StakeAction): ActionValue {
+      return new ActionValue(ActionKind.STAKE, changetype<u32>(input))
+    }
+
+    static fromAddKey(input: AddKeyAction): ActionValue {
+      return new ActionValue(ActionKind.ADD_KEY, changetype<u32>(input))
+    }
+
+    static fromDeleteKey(input: DeleteKeyAction): ActionValue {
+      return new ActionValue(ActionKind.DELETE_KEY, changetype<u32>(input))
+    }
+
+    static fromDeleteAccount(input: DeleteAccountAction): ActionValue {
+      return new ActionValue(ActionKind.DELETE_ACCOUNT, changetype<u32>(input))
+    }
+  }
+
+  // We don't map ReceiptData
+  export class ActionReceipt {
+    constructor(
+      // Receipt fields
+      public predecessorId: string,
+      public receiverId: string,
+      public id: CryptoHash,
+
+      // ReceiptAction fields
+      public signerId: string,
+      public signerPublicKey: PublicKey,
+      public gasPrice: BigInt,
+      public outputDataReceivers: Array<DataReceiver>,
+      public inputDataIds: Array<CryptoHash>,
+      public actions: Array<ActionValue>,
+    ) {}
+  }
+
+  export enum SuccessStatusKind {
+    VALUE = 0,
+    RECEIPT_ID = 1,
+  }
+
+  // Doesn't have Value suffix because it has
+  // VALUE variant/kind, that would be confusing.
+  export class SuccessStatus {
+    constructor(
+      public kind: SuccessStatusKind,
+      public data: Payload,
+    ) {}
+
+    toValue(): string {
+      assert(this.kind == SuccessStatusKind.VALUE, 'SuccessStatus is not a \'Value\'.')
+      return changetype<string>(this.data as u32)
+    }
+
+    toReceiptId(): CryptoHash {
+      assert(this.kind == SuccessStatusKind.RECEIPT_ID, 'SuccessStatus is not a \'ReceiptId\'.')
+      return changetype<CryptoHash>(this.data as u32)
+    }
+
+    static fromValue(input: string): SuccessStatus {
+      return new SuccessStatus(SuccessStatusKind.VALUE, changetype<u32>(input))
+    }
+
+    static fromReceiptId(input: CryptoHash): SuccessStatus {
+      return new SuccessStatus(SuccessStatusKind.RECEIPT_ID, changetype<u32>(input))
+    }
+  }
+
+  export enum Direction {
+    LEFT = 0,
+    RIGHT = 1,
+  }
+
+  export class MerklePathItem {
+    constructor(
+      public hash: CryptoHash,
+      public direction: Direction,
+    ) {}
+
+    @operator('<')
+    lt(other: MerklePathItem): boolean {
+      abort("Less than operator isn't supported in MerklePathItem")
+      return false
+    }
+
+    @operator('>')
+    gt(other: MerklePathItem): boolean {
+      abort("Greater than operator isn't supported in MerklePathItem")
+      return false
+    }
+
+    toString(): string {
+      return `{hash: ${this.hash.toString()}}, direction: ${this.direction.toString()}`
+    }
+  }
+
+  export class MerklePath extends Array<MerklePathItem> {}
```

Comment:
Could we actually get rid of `MerklePath` here, I'm not seeing the added benefits of having it. Maybe there is something else in play that I do not see.

Reply:
In fact, I'm wondering what is the correct Rust bindings for this one, is it `AscPtr<AscMerklePath>` or directly `AscPtr<Array<AsckMerklePathItem>>`?

Reply:
I think we want to have the first one because it will be a new class, and that requires a new class identifier for the header.

Reply:
I'll still wondering how such "extended" array should look like when generating the underlying `AscMerklePath` here. How to represent the "extends" fact. Will need you help to determine the correct mapping.